subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemSupport.java

```diff
@@ -232,6 +242,30 @@ public void execute(CompatibilityCheckDetails<LibraryElements> details) {
         }
     }
 
+    public static class TargetJvmEnvironmentCompatibilityRules implements AttributeCompatibilityRule<TargetJvmEnvironment>, ReusableAction {
+
+        @Override
+        public void execute(CompatibilityCheckDetails<TargetJvmEnvironment> details) {
+            details.compatible();
+        }
+    }
+
+    public static class TargetJvmEnvironmentDisambiguationRules implements AttributeDisambiguationRule<TargetJvmEnvironment>, ReusableAction {
+
+        @Override
+        public void execute(MultipleCandidatesDetails<TargetJvmEnvironment> details) {
+            List<TargetJvmEnvironment> exactMatches = details.getCandidateValues().stream().filter(c -> c == details.getConsumerValue()).collect(Collectors.toList());
```

Comment:
I find the functional style hard to read here.

Why not two `contains` instead?
Would read better IMO.

Reply:
It looked better in Kotlin where I prototyped this. 😅 

I can change it. I don't mind either way.

Reply:
Mh. How do you mean to use `contains`. The only non functional style I can think of right now is looping though the whole list to find out that there is **exactly one** detail matching the condition. But maybe I just don't see it.

Reply:
`candidatesValues` is a `Set`, so either it `contains` the `consumerValue` or it does not.
And similarly for whether or not it `contains` `STANDARD_JVM`.
Or maybe I am the one missing something.

Reply:
Ah yes. Sorry my mistake. We disambiguate the attributes, not the complete variants. So it is a Set. And there can only be one or none. Thanks for elaborating.

I improved the code.